lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -89,6 +89,12 @@ static CodableConformanceType typeConformsToCodable(TypeChecker &tc,
   // Some generic types need to be introspected to get at their "true" Codable
   // conformance.
   auto canType = target->getCanonicalType();
+  if (auto referenceType = dyn_cast<ReferenceStorageType>(canType)) {
```

Comment:
@jrose-apple I guess that's a question I have here — thinking about it, is getting the canonical type actually necessary here? We just want to unwrap layers of types (i.e. strip optionality, ownership semantics, etc.) to get at the inner type, then check if that inner type conforms to `Encodable` or `Decodable`. Since I'm guessing that `TypeChecker::conformsToProtocol` will already do the appropriate lookup, are there cases here where it's even necessary to get the canonical type?